flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala

```diff
@@ -467,6 +467,9 @@ object FlinkTypeFactory {
       val mapRelDataType = relDataType.asInstanceOf[MapRelDataType]
       mapRelDataType.mapType
 
+    // CURSOR for UDTF case, whose type info will never be used, just a placeholder
+    case CURSOR => NothingType.INSTANCE
```

Comment:
Just use a `GenericType(NothingTypeInfo)` is ok?

Reply:
make sense.